pulp_container/app/registry_api.py

```diff
@@ -349,6 +324,24 @@ def get_dr_push(self, request, path, create=False):
                 raise RepositoryNotFound(name=path)
         return distribution, repository
 
+    def create_dr(self, path, request):
+        try:
+            with transaction.atomic():
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+                distribution = serializers.ContainerDistributionSerializer.get_or_create(
+                    {"base_path": path, "name": path}, {"repository": get_url(repository)}
+                )
+        except ObjectDoesNotExist:
+            raise RepositoryInvalid(name=path)
+
+        repository = repository.cast()
+        if not repository.PUSH_ENABLED:
```

Comment:
yeah so this needs to be moved upper per my previous comment. now you're checking the repo you have created on line 330